components/vni.js

```diff
@@ -0,0 +1,63 @@
+// Vni.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+var inputs = require('./input-state');
+
+// Hash of vnis
+vnis = []; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function Vni( vnid ) {
+  this.vnid = vnid;
+  this.lm = Lm();
+  this.allInputStates = inputs.allInputStates;
+  this.inputState = inputs.inputState;
+  this.output = {};
+  this.error = {};
+  this.node = this.nodeInstance;
```

Comment:
I prefer to only include code that is needed to complete this sprint and not sprinkle code with features that have not been fully discussed/fleshed out.